common/openapi.yml

```diff
@@ -759,6 +759,9 @@ components:
             string
           format: uuid
           example: 00000000-0000-0000-0000-000000000000
+        engine_id:
```

Comment:
Not that I hate this here, but it feels odd that we are relying on external API types for internal data passing.

I think I'd rather change the return types of the internal methods rather than change the external API, if we can?